src/Compilers/CSharp/Portable/CSharpParseOptions.cs

```diff
@@ -228,6 +173,32 @@ public new CSharpParseOptions WithFeatures(IEnumerable<KeyValuePair<string, stri
             }
         }
 
+        internal override void ValidateOptions(ArrayBuilder<Diagnostic> builder)
+        {
+            ValidateOptions(builder, MessageProvider.Instance);
+
+            // Validate LanguageVersion not SpecifiedLanguageVersion, after Latest/Default has been converted:
+            if (!LanguageVersion.IsValid())
```

Comment:
Discussed offline. Agreed that we shouldn't throw at all.

---
In reply to: [101868422](https://github.com/dotnet/roslyn/pull/17113#discussion_r101868422) [](ancestors = 101868422)